src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/configmaps.tf

```diff
@@ -30,6 +30,11 @@ locals {
 resource "local_file" "jupyter_server_config_py" {
   content  = local.jupyter-notebook-config-py-template
   filename = "${path.module}/files/jupyter/jupyter_server_config.py"
+
+  provisioner "local-exec" {
+    # check the syntax of the config file without running it
+    command = "python -m py_compile ${self.filename}"
```

Comment:
So this was able to detect the syntax error, but it did not stop deployment:

```
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py: Creating...
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py: Provisioning with 'local-exec'...
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py (local-exec): Executing: ["/bin/sh" "-c" "python -m py_compile modules/kubernetes/services/jupyterhub/files/jupyter/jupyter_server_config.py"]
[terraform]: module.jupyterhub.local_file.overrides_json: Creating...
[terraform]: module.jupyterhub.local_file.overrides_json: Creation complete after 0s [id=5d5e4f93190b880f9778b9d4487639fdc8406a2c]
[terraform]: module.jupyterhub.random_password.proxy_secret_token: Creating...
[terraform]: module.kubernetes-nfs-server[0].kubernetes_deployment.main: Creating...
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py (local-exec):   File "modules/kubernetes/services/jupyterhub/files/jupyter/jupyter_server_config.py", line 13
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py (local-exec):     preferred_dir =
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py (local-exec):                    ^
[terraform]: module.jupyterhub.local_file.jupyter_server_config_py (local-exec): SyntaxError: invalid syntax
[terraform]: 
[terraform]: module.forwardauth.module.forwardauth-openid-client.random_password.client_secret: Creating...
[terraform]: module.jupyterhub.random_password.proxy_secret_token: Creation complete after 0s [id=none]
```

Reply:
Oh, actually it worked:

```
[terraform]: ╷
[terraform]: │ Error: local-exec provisioner error
[terraform]: │ 
[terraform]: │   with module.jupyterhub.local_file.jupyter_server_config_py,
[terraform]: │   on modules/kubernetes/services/jupyterhub/configmaps.tf line 34, in resource "local_file" "jupyter_server_config_py":
[terraform]: │   34:   provisioner "local-exec" {
[terraform]: │ 
[terraform]: │ Error running command 'python -m py_compile
[terraform]: │ modules/kubernetes/services/jupyterhub/files/jupyter/jupyter_server_config.py':
[terraform]: │ exit status 1. Output:   File
[terraform]: │ "modules/kubernetes/services/jupyterhub/files/jupyter/jupyter_server_config.py",
[terraform]: │ line 13
[terraform]: │     preferred_dir =
[terraform]: │                    ^
[terraform]: │ SyntaxError: invalid syntax
[terraform]: │ 
[terraform]: │ 
[terraform]: ╵
```

This behaviour is explained in detail in https://github.com/hashicorp/terraform/issues/21088:

> `on_failure = "fail"` is the default behavior, and the resource will be left in a "tainted" state. This also produces an error, which will prevent dependent resources from being applied. This will not halt other operations already in progress, since they must be independent of the resource being provisioned.

So this is all good, it will indeed capture problems before we merge any other changes in the future.